src/big_query.ml

```diff
@@ -132,6 +132,55 @@ module Datasets = struct
       Error.of_response_status_code_and_body x body
 end
 
+module Tables = struct
+
+  let list ?project_id ?max_results ?page_token ~dataset_id () : (string, [> Error.t ]) Lwt_result.t =
+    let open Lwt_result.Infix in
+
+    Auth.get_access_token ~scopes:[Scopes.bigquery] ()
+    |> Lwt_result.map_err (fun e -> `Gcloud_auth_error e)
+    >>= fun token_info ->
+
+    let project_id =
+      project_id |> CCOpt.get_or ~default:token_info.project_id
+    in
+
+    Lwt.catch (fun () ->
+        let uri = Uri.make ()
+            ~scheme:"https"
+            ~host:"www.googleapis.com"
+            ~path:(Printf.sprintf "bigquery/v2/projects/%s/datasets/%s/tables" project_id dataset_id)
```

Comment:
I know we haven't been sticking to it so far but I think it'd be good to have the modules follow the API URL structure, so `Datasets.Tables.list`